Compiled Nextstrain SARS-CoV-2 resources are available at [nextstrain.org/sars-cov-2](https://nextstrain.org/sars-cov-2/). Follow [@nextstrain](https://twitter.com/nextstrain) for continual data updates.

This phylogeny shows evolutionary relationships of SARS-CoV-2 viruses from the ongoing COVID-19 pandemic. Although the genetic relationships among sampled viruses are quite clear, there is considerable uncertainty surrounding estimates of specific transmission dates and in reconstruction of geographic spread. Please be aware that specific inferred geographic transmission patterns and temporal estimates are only a hypothesis.
The analysis on this page uses data from completely open sources, such that we can make input data and intermediate files available for further analysis (see below).

There are hundreds of thousands of complete SARS-CoV-2 genomes available on open databases and this number increases every day, but geographical representation varies. This visualization can only handle ~3000 genomes in a single view for performance and legibility reasons. Because of this we subsample available genome data for these analysis views. Our primary [global analysis](/ncov/open/global/) subsamples to ~600 genomes per continental region with ~400 from the previous 4 months and ~200 from before this. This results in a more equitable global sequence distribution, but hides samples available from regions that are doing lots of sequencing. To mitigate against this, we've set up separate analyses to focus on particular regions. They are available on the "Dataset" dropdown on the left or by clicking on the following links: [Africa](/ncov/open/africa?f_region=Africa), [Asia](/ncov/open/asia?f_region=Asia), [Europe](/ncov/open/europe?f_region=Europe), [North America](/ncov/open/north-america?f_region=North%20America), [Oceania](/ncov/open/oceania?f_region=Oceania) and [South America](/ncov/open/south-america?f_region=South%20America).
But be aware that not all regions are well represented in open data bases.

Site numbering and genome structure uses [Wuhan-Hu-1/2019](https://www.ncbi.nlm.nih.gov/nuccore/MN908947) as reference. The phylogeny is rooted relative to early samples from Wuhan. Temporal resolution assumes a nucleotide substitution rate of 8 &times; 10^-4 subs per site per year. Full details on bioinformatic processing can be found [here](https://github.com/nextstrain/ncov).

We gratefully acknowledge the authors, originating and submitting laboratories of the genetic sequence and metadata for sharing their work in open data bases. A full listing of all originating and submitting laboratories is available below. An attribution table is available by clicking on "Download Data" at the bottom of the page and then clicking on "Strain Metadata" in the resulting dialog box.

To maximize the utility and visibility of these generously shared data, we provide preprocessed files that can serve as a starting point for additional analysis.

### All sequences, the full alignment, and metadata

 * [sequences.fasta.xz]()
 * [alignment.fasta.xz]()
 * [metadata.tsv.xz]()

 ### Subsampled sequences and intermediate files

 * [alignment.fasta.xz]()
 * [metadata.tsv.xz]()
 * [tree.nwk]()

